world-chain-builder/crates/world/pool/src/tx.rs

```diff
@@ -14,31 +14,36 @@ use reth_primitives::{kzg::KzgSettings, Recovered};
 use reth_primitives_traits::InMemorySize;
 use revm_primitives::{Address, TxKind, B256, U256};
 use thiserror::Error;
-use world_chain_builder_pbh::payload::PbhValidationError;
+use world_chain_builder_pbh::payload::{PBHPayload, PbhValidationError};
 
 #[derive(Debug, Clone)]
 pub struct WorldChainPooledTransaction {
     pub inner: OpPooledTransaction,
-    pub valid_pbh: bool,
+    pub payload: Option<Vec<PBHPayload>>,
 }
 
 pub trait WorldChainPoolTransaction: EthPoolTransaction {
     fn valid_pbh(&self) -> bool;
-    fn set_valid_pbh(&mut self);
+    fn set_pbh_payloads(&mut self, payload: Vec<PBHPayload>);
     fn conditional_options(&self) -> Option<&TransactionConditional>;
+    fn payload(&self) -> Option<&Vec<PBHPayload>>;
```

Comment:
Wdyt about updating to `pbh_payload`? Within the builder logic, this makes it a little more clear that `tx.pbh_payload()` is checking for the PBH data compared to `tx.payload()`.